transport/src/main/java/io/netty/bootstrap/ChannelInitializerExtension.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.bootstrap;
+
+import io.netty.channel.Channel;
+
+/**
+ * A channel initializer extension make it possible to enforce rules and apply modifications across multiple,
+ * disconnected uses of Netty within the same JVM process.
+ * <p>
+ * For instance, application-level firewall rules can be injected into all uses of Netty within an application,
+ * without making changes to such uses that are otherwise outside the purview of the application code,
+ * such as 3rd-party libraries.
+ * <p>
+ * All channel initializer extensions that are available on the classpath will be
+ * {@linkplain java.util.ServiceLoader#load(Class) service-loaded} and used by all {@link AbstractBootstrap} subclasses,
+ * unless the extensions are explicitly disabled on the given bootstrap instance with a call to
+ * {@link AbstractBootstrap#disableChannelInitializerExtensions()}.
+ */
+public interface ChannelInitializerExtension {
```

Comment:
SInce netty 4.1 is pre java8 It would be impossible to add new methods in backward compatible way. Maybe declare this as abstract class?

Reply:
@mostroverkhov is right... I think we should use an abstract class